packages/realm/src/app-services/SyncConfiguration.ts

```diff
@@ -366,7 +366,7 @@ function toSSLVerifyCallbackWithListArguments(verifyCallback: SSLVerifyCallback)
 function parseClientResetConfig(clientReset: ClientResetConfig | undefined, onError: ErrorCallback | undefined) {
   if (!clientReset) {
     return {
-      clientResyncMode: undefined,
+      clientResyncMode: toBindingClientResetMode(ClientResetMode.RecoverOrDiscardUnsyncedChanges),
```

Comment:
```suggestion
      clientResyncMode: toBindingClientResetMode(ClientResetMode.RecoverUnsyncedChanges),
```
I have changed my opinion on this.  Perhaps having `discard` as the default is too destructive.  Let's align with the documentation and other SDKs:
https://www.mongodb.com/docs/atlas/app-services/sync/error-handling/client-resets/#recover-unsynced-changes